userspace/libsinsp/filter/parser.cpp

```diff
@@ -591,7 +591,65 @@ inline const char* parser::cursor()
 	return m_input.c_str() + m_pos.idx;
 }
 
-string parser::escape_str(const string& str)
+std::string libsinsp::filter::escape_str(const std::string& str)
+{
+	std::string res = "";
+	size_t len = str.size();
+	bool should_escape = false;
+	for (size_t i = 0; i < len; i++)
+	{
+		switch(str[i])
+		{
+		case '\b':
+			should_escape = true;
+			res += "\\b";
+			break;
+		case '\f':
+			should_escape = true;
+			res += "\\f";
+			break;
+		case '\n':
+			should_escape = true;
+			res += "\\n";
+			break;
+		case '\r':
+			should_escape = true;
+			res += "\\r";
+			break;
+		case '\t':
+			should_escape = true;
+			res += "\\t";
+			break;
+		case ' ':
+			should_escape = true;
+			res += ' ';
+			break;
+		case '\\':
+			should_escape = true;
+			res += "\\\\";
+			break;
+		case '"':
+			should_escape = true;
+			res += "\\\"";
+			break;
+		case '\'':
+			should_escape = true;
+			res += "\\\"";
```

Comment:
```suggestion
			res += "'";
```
We should just go for this at this point. `'` is an allowed character if we escape by `"`.
The goal is to enforce the `"` escaping delimiter, but not to change the original string. WDYT?